src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -323,6 +366,8 @@ public Task<ODataBatchOperationRequestMessage> CreateOperationRequestMessageAsyn
         public ODataBatchOperationResponseMessage CreateOperationResponseMessage(string contentId)
         {
             this.VerifyCanCreateOperationResponseMessage(true);
+            Debug.Assert(this.currentOperationContentId == null, "The Content-ID header is only supported in request messages.");
+
             return this.CreateOperationResponseMessageImplementation(contentId);

```

Comment:
Really?  In json implementation we pass it in to BuildOperationResponseMessage (as we should), and in multipartmixed we *should* pass it in to BuildOperationResponseMessage and *do* use it when writing the response preamble.

Reply:
yes, that is the current implementation.
[https://github.com/OData/odata.net/blob/75df8f44f2b81f984589790be4885b6ee8946ad0/src/Microsoft.OData.Core/ODataBatchOperationResponseMessage.cs#L164](https://github.com/OData/odata.net/blob/75df8f44f2b81f984589790be4885b6ee8946ad0/src/Microsoft.OData.Core/ODataBatchOperationResponseMessage.cs#L164)
, where the response object is created with /*contentId*/ null.
I think it is due to the fact that contentId in the response is not much useful comparing to that in the request?
I agree that it doesn't hurt to add it. The question is whether we have to do this now.
thoughts?


---
In reply to: [155392178](https://github.com/OData/odata.net/pull/992#discussion_r155392178) [](ancestors = 155392178)

Reply:
Yes, we should remove the assert and, in multiparmixedbatchwriter.CreationOperationResponseMessageImplementation, we should pass the contentId in to BuildOoperationResponseMessage.

---
In reply to: [155410202](https://github.com/OData/odata.net/pull/992#discussion_r155410202) [](ancestors = 155410202,155392178)

Reply:
OK, done. 
Shouldn't hurt to add, as long as we don't remove the contentId later.

---
In reply to: [156776160](https://github.com/OData/odata.net/pull/992#discussion_r156776160) [](ancestors = 156776160,155410202,155392178)